### PR TITLE
Fix creation events not sent for implicit creation of object in the default bucket (fixes #529)

### LINKED ISSUE #529: Create Collection event notification not fired for auto-create on default bucket
I was hacking around with Redis event notifications today. I setup two separate listeners with filters:

```
kinto.event_listeners = redis1 redis2

kinto.event_listeners.redis1.use = cliquet.listeners.redis
kinto.event_listeners.redis1.url = <url>
kinto.event_listeners.redis1.pool_size = 5
kinto.event_listeners.redis1.listname = kinto:collection:create
kinto.event_listeners.redis1.actions = create
kinto.event_listeners.redis1.resources = collection

kinto.event_listeners.redis2.use = cliquet.listeners.redis
kinto.event_listeners.redis2.url = <url>
kinto.event_listeners.redis2.pool_size = 5
kinto.event_listeners.redis2.listname = kinto:record:create
kinto.event_listeners.redis2.actions = create
kinto.event_listeners.redis2.resources = record
```

To test, I was using the default bucket, and the inserting a sample record to a collection that did not yet exist. Although the collection was auto-created, and the record inserted, only the event for create record was logged to Redis. If I create the collection manually, then the create collection event is logged.

Summary: Auto-created collections on the default bucket are not firing event notifications.

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

File: kinto/plugins/default_bucket/__init__.py [hunk 1]
```diff
@@ -73,8 +80,14 @@ def create_collection(request, bucket_id):
...
         raise_invalid(request, **error_details)
+    data = {'id': collection_id}
     try:
-        collection = resource.model.create_record({'id': collection_id})
+        collection = resource.model.create_record(data)
+        # Since the current request is not a resource (but a straight Service),
+        # we simulate a request on a bucket resource.
+        # This will be used in the resource event payload.
+        request.current_resource_name = 'collection'
+        resource.postprocess(data, action=ACTIONS.CREATE)
```
Comment (almet) [inline kinto/plugins/default_bucket/__init__.py]: Rather than duplicating, we should factorize this code.

Comment (leplatrem) [inline kinto/plugins/default_bucket/__init__.py]: You mean here in this file ?

Comment (almet) [inline kinto/plugins/default_bucket/__init__.py]: I see twice the same code here in this file, so having a helper for it should make it easier to maintain.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -9,6 +9,8 @@ This document describes changes between each past release.
 **Bug fixes**
 
 - Relax content-type validation when no body is posted (fixes #507)
+- Fix creation events not sent for implicit creation of object in the ``default``
```
Comment (Natim) [inline CHANGELOG.rst]: of objects

File: kinto/plugins/default_bucket/__init__.py (hunk 1 above)
Comment (Natim) [inline kinto/plugins/default_bucket/__init__.py]: Do you mean something like:

```
def resource_handling(resource, request, data, resource_name, resource_id):
    data = {'id': resource_id}
    obj = resource.model.create_record(data)
    request.current_resource_name = resource_name
    resource.postprocess(data, action=ACTIONS.CREATE)
    return obj
```

**Comment by Natim** [comment]
LGTM

File: kinto/plugins/default_bucket/__init__.py (hunk 1 above)
Comment (almet) [inline kinto/plugins/default_bucket/__init__.py]: That's what I mean, yes!

**Pull request merged**